programs/mp-sol-restaking/src/instructions/users/unstake.rs

```diff
@@ -19,6 +19,9 @@ pub struct Unstake<'info> {
 
     #[account(mut)]
     pub mpsol_mint: Box<Account<'info, Mint>>,
+    #[account(mut)]
+    /// CHECK: compare to set acc in main state
```

Comment:
Please make it optional account and check owner/layout in code

Reply:
we need the withdrawal_fee, so in all cases it should not be None
I've added a check to avoid failing the txn if the admin set a bad treasury account